src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -357,47 +369,159 @@ def executeMcpTestSteps(self, config, test_type):
       self._sas_admin.InjectPeerSas({'certificateHash': certificate_hash,
                                      'url': sas_test_harness_object.getBaseUrl()})
       sas_test_harness_objects.append(sas_test_harness_object)
-
-
+   
     # Step 4,5 : Inject IAP protected entities into UUT
     for iteration_content in config['iterationData']:
-
       # Execute steps for single iteration
-      self.executeSingleMCPIteration(test_type, iteration_content, sas_test_harness_objects,
+      self.executeSingleMCPIteration(test_type, iteration_content, sas_test_harness_objects, 
                                      domain_proxy_objects)
-
+    
     # Stopping Test harness servers
     for test_harness in sas_test_harness_objects:
       test_harness.shutdown()
       del test_harness
 
-  def checkMcpIap(self, protected_entity, agg_interference_list, iap_refmodel_margin_list, delta_iap):
-    """Calculating IAP for each FSS and ESC sensor protection point,
-    and for at least 95% of the protection points of each PPA and GWPZ.
-    """
-    pass
+  def performIapAndAggregateInterferenceCheck(self, protection_entities, sas_uut,
+                                      sas_th_fad_objects, cbsd_list):
+    """Compare the interference calculated using iap and aggregate reference models.
 
-  def callAggregateInterferenceModel(self, cbsd_list, protected_entity):
-    """Args:
-         cbsd_list: List of cbsds with Authorized grants
-         protected_entity: A dictionary with multiple key-value pairs where
-         keys are protection entities
+    Args:
+      protection_entities : List of entity records needed to be protected 
+      sas_uut : FAD object from SAS UUT
+      sas_th_fad_objects : a list of FAD objects from SAS Test Harness
+      cbsd_list : List of new cbsds registered to the SAS.
+
+    Returns:
+      True or False based on the comparison of Aggregate Interference values
+      against the two reference models
+    """
+    
+    # Calculate and compare the interference value for PPA protection entity 
+    if 'ppaRecords' in protection_entities:
+      for ppa_record in protection_entities['ppaRecords']:
+        pal_records = protection_entities['palRecords']
+        pool = Pool(processes=min(multiprocessing.cpu_count(), NUM_OF_PROCESS))
+        # Call iap reference model for ppa
+        ppa_ap_iap_ref_values = iap.\
+             performIapForPpa(ppa_record, sas_uut, sas_th_fad_objects, pal_records, pool)
```

Comment:
remove last argument here and in following line

Reply:
removed